_inc/client/plans/single-product-search/index.jsx

```diff
@@ -1,22 +1,60 @@
 /**
  * External dependencies
  */
-import React from 'react';
-import { translate as __ } from 'i18n-calypso';
-import { get, noop } from 'lodash';
+import React, { useState } from 'react';
+import { connect } from 'react-redux';
+import { translate as __, numberFormat } from 'i18n-calypso';
+import { get } from 'lodash';
 
 /**
  * Internal dependencies
  */
 import Button from 'components/button';
 import PlanRadioButton from '../single-product-components/plan-radio-button';
+import ProductSavings from '../single-product-components/product-savings';
+import {
+	JETPACK_SEARCH_TIER_UP_TO_100_RECORDS,
+	JETPACK_SEARCH_TIER_UP_TO_1K_RECORDS,
+	JETPACK_SEARCH_TIER_UP_TO_10K_RECORDS,
+	JETPACK_SEARCH_TIER_UP_TO_100K_RECORDS,
+	JETPACK_SEARCH_TIER_UP_TO_1M_RECORDS,
+	JETPACK_SEARCH_TIER_MORE_THAN_1M_RECORDS,
+} from '../../lib/plans/constants';
+import { getUpgradeUrl } from '../../state/initial-state';
 
-const PLACEHOLDER_RECORD_COUNT = 53;
-const PLACEHOLDER_PRICE = 50;
-const PLACEHOLDER_LABEL = 'Tier 1: Up to 100 records';
+function getTierLabel( priceTierSlug, recordCount ) {
+	switch ( priceTierSlug ) {
+		case JETPACK_SEARCH_TIER_UP_TO_100_RECORDS:
+			return __( 'Up to 100 records' );
+		case JETPACK_SEARCH_TIER_UP_TO_1K_RECORDS:
+			return __( 'Up to 1,000 records' );
+		case JETPACK_SEARCH_TIER_UP_TO_10K_RECORDS:
+			return __( 'Up to 10,000 records' );
+		case JETPACK_SEARCH_TIER_UP_TO_100K_RECORDS:
+			return __( 'Up to 100,000 records' );
+		case JETPACK_SEARCH_TIER_UP_TO_1M_RECORDS:
+			return __( 'Up to 1,000,000 records' );
+		case JETPACK_SEARCH_TIER_MORE_THAN_1M_RECORDS:
+			const tierMaximumRecords = 1000000 * Math.ceil( recordCount / 1000000 );
+			return __( 'Up to %(tierMaximumRecords)s records', {
+				args: { tierMaximumRecords: numberFormat( tierMaximumRecords ) },
+			} );
+		default:
+			return null;
+	}
+}
+
+function handleSelectedTimeframeChangeFactory( setTimeframe ) {
+	return event => setTimeframe( event.target.value );
+}
 
-export default function SingleProductSearchCard( props ) {
-	const currencyCode = get( props.products, 'jetpack_backup_daily.currency_code', '' );
+export function SingleProductSearchCard( props ) {
+	const [ timeframe, setTimeframe ] = useState( 'yearly' );
+	const handleSelectedTimeframeChange = handleSelectedTimeframeChangeFactory( setTimeframe );
+	const currencyCode = get( props.products, 'jetpack_search.currency_code', '' );
+	const monthlyPrice = get( props.products, 'jetpack_search_monthly.cost', '' );
+	const yearlyPrice = get( props.products, 'jetpack_search.cost', '' );
```

Comment:
I think you need to get all these properties from `props.siteProducts` rather than `props.products` (in fact, `props.products` can probably be removed from this component entirely)?

`siteProducts` is the one that has the site-specific pricing (based on the current search tier) so without that I think the wrong price will be shown.

Reply:
Gotcha, I'll make the fix.